primitives/consensus/babe/src/digests.rs

```diff
@@ -18,178 +18,80 @@
 
 #[cfg(feature = "std")]
 use super::{BABE_ENGINE_ID, AuthoritySignature};
-#[cfg(not(feature = "std"))]
-use super::{VRF_OUTPUT_LENGTH, VRF_PROOF_LENGTH};
 use super::{AuthorityId, AuthorityIndex, SlotNumber, BabeAuthorityWeight};
+use sp_core::RuntimeDebug;
 #[cfg(feature = "std")]
 use sp_runtime::{DigestItem, generic::OpaqueDigestItemId};
 #[cfg(feature = "std")]
 use std::fmt::Debug;
 use codec::{Decode, Encode};
 #[cfg(feature = "std")]
-use codec::{Codec, Input, Error};
-#[cfg(feature = "std")]
-use schnorrkel::{
-	SignatureError, errors::MultiSignatureStage,
-	vrf::{VRFProof, VRFOutput, VRF_OUTPUT_LENGTH, VRF_PROOF_LENGTH}
-};
+use codec::Codec;
+use sp_consensus_vrf::schnorrkel::{VRFProof, VRFOutput, Randomness};
 use sp_std::vec::Vec;
 
+/// A BABE primary pre-runtime digest;
+#[derive(Clone, RuntimeDebug, Encode, Decode)]
+pub struct PrimaryPreDigest {
+	/// VRF output
+	pub vrf_output: VRFOutput,
+	/// VRF proof
+	pub vrf_proof: VRFProof,
+	/// Authority index
+	pub authority_index: AuthorityIndex,
+	/// Slot number
+	pub slot_number: SlotNumber,
+}
+
+/// A BABE secondary pre-runtime digest.
+#[derive(Clone, RuntimeDebug, Encode, Decode)]
+pub struct SecondaryPreDigest {
+	/// Authority index
```

Comment:
```suggestion
	/// Authority index
	///
	/// This is not strictly necessary, since the secondary slots
	/// are assigned based on slot number and epoch randomness. But including
	/// it makes things easier for higher-level users of the chain data to
	/// be aware of the author of a secondary-slot block.
```